datadog-tracer-flare/src/lib.rs

```
@@ -0,0 +1,176 @@
+        language,
+        tracer_version,
+        endpoint: remote_config_endpoint,
+        products: vec![
+            RemoteConfigProduct::AgentConfig,
+            RemoteConfigProduct::AgentTask,
+        ],
+        capabilities: vec![],
+    };
+
+    SingleChangesFetcher::new(
+        ParsedFileStorage::default(),
+        Target {
+            service,
+            env,
+            app_version,
+            tags: vec![],
+        },
+        runtime_id,
+        config_to_fetch,
+    )
+}
+
+/// Function that listen to RemoteConfig on the agent
+///
+/// # Arguments
+///
+/// * `listener` - Listener use to fetch RemoteConfig from the agent with specific config.
+///
+/// # Returns
+///
+/// * `Ok()` - If successful.
+/// * `FlareError(msg)` - If something fail.
+#[allow(clippy::too_many_arguments)]
+pub async fn run_remote_config_listener(
+    listener: &mut SingleChangesFetcher<RawFileStorage<Result<RemoteConfigData, anyhow::Error>>>,
+) -> Result<(), FlareError> {
+    match listener.fetch_changes().await {
+        Ok(changes) => {
+            println!("Got {} changes.", changes.len());
+            for change in changes {
+                match change {
+                    Change::Add(file) => {
+                        println!("Added file: {} (version: {})", file.path(), file.version());
+                        println!("Content: {:?}", file.contents().as_ref());
+                    }
+                    Change::Update(file, _) => {
+                        println!(
+                            "Got update for file: {} (version: {})",
+                            file.path(),
+                            file.version()
+                        );
+                    }
+                    Change::Remove(file) => {
+                        println!("Removing file {}", file.path());
+                    }
+                }
+            }
+        }
+        Err(e) => {
+            eprintln!("Fetch failed with {e}");
```

Should we be returning an error here instead of just swallowing it?

---

Yes, done